### ISSUE TITLE
fopen() failed to open stream on temp/ directory

### ISSUE THREAD

**Issue by RobErskine** [nystudio107/craft3-imageoptimize#17]
Hey there. Awesome plugin. I'm looking to jump into some of the more robust features of this plugin but am running into some errors.

I'm fairly new to Craft 3 so I'm still getting accustomed to how Craft handles temporary assets and transforms. 

I'm getting the following error when trying to save out assets:

![fopen() failed](https://cl.ly/0i051W0c1f38/Screen%20Shot%202017-10-24%20at%204.11.35%20PM.jpg)

Likewise when I try to use the webp transform, I get the following error:

![filesize() stat failed](https://cl.ly/270n1o3j2A2r/Screen%20Shot%202017-10-24%20at%204.24.20%20PM.jpg)

Clearly it has to do with Craft not being able to read from my template directory. I've adjusted permissions to `777` to test, as well as `chown`ing the folder to my root username to no avail.

Some more information on my setup: I'm using remote asset volumes through Rackspace Cloud Files. Before I attached an OptimizedFiles field to my Asset field I'm able to save out assets just fine. My assets _are_ being saved to the `/storage/runtime/temp/` directory, but I guess they're just not being read.

Not sure if this is an error with the plugin, or an error with my setup, but would love some guidance and an opportunity to contribute to the instructions to make sure no one else gets caught with this in their transition from Craft 2 -> Craft 3.

**Comment by khalwat** [comment]
We might need to do some screen sharing to look at this setup. Hit me up on Slack?

**Comment by khalwat** [comment]
Could also be a bug with the Rackspace Cloud Files plugin?

**Comment by RobErskine** [comment]
I thought of that too. I changed it from a Rackspace Cloud Files volume to a local volume and the issue still persisted. I'll hit you up on Slack!

**Comment by khalwat** [comment]
Any luck @RobErskine ?

**Comment by RobErskine** [comment]
Yes, sorry I forgot to post back with how I netted out.

tl;dr: My optim library paths were messed up.

@superflausch hit me up on Slack and told me he was having my same problem. After some debugging, he noticed that it was related to not having JPG Optim installed on his local machine. After commenting out the Active Image Processors, and Active Image Variant Creators, I was able to upload an image and get it to generate variants. The issue wasn't with the plugin,  but my processor libs. We were thinking it might be advantageous if there was a check if those processors were available as part of the plugin. But that's more of a FR. Anyway, I'm all good now. Thanks for your help!

```
    // Active image processors
    'activeImageProcessors'      => [

    ],

    // Active image variant creators
    'activeImageVariantCreators' => [

    ],
```
